Add feature: image comparison with slider

joshjavier opened the pull request:
I replaced `<img id="imgPreview">` with an `<image-compare>` web component that has a slider for comparing the original and compressed versions of an image.

Here's how it looks (the quality is set to 0 to make the distinction between the original and compressed versions more obvious):

![](https://github.com/joshjavier/image-compressroar/assets/46987872/ec897571-e386-476c-bb2c-27ad1a156142)

This imports the [@cloudfour/image-compare](https://www.npmjs.com/package/@cloudfour/image-compare) npm package to the `main.js` bundle. [Full documentation here.](https://image-compare-component.netlify.app/)

---

For context, I also tried [another image compare component](https://image-compare-viewer.netlify.app/) but although it has more customization options, it is not accessible and has a [larger bundle size](https://bundlephobia.com/package/image-compare-viewer@1.6.2) (3.3 kB) than [image-compare](https://bundlephobia.com/package/@cloudfour/image-compare@1.0.5) (1.6 kB).

joshjavier committed:
Add @cloudfour/image-compare as a dependency

joshjavier committed:
Replace image preview with <image-compare>

johnllyod commented:
The image compare feature is working well but for some reason the "download compressed image" button is not working on my end.
![Can't download image](https://github.com/joshjavier/image-compressroar/assets/48270436/35d581dd-5ad2-4abe-bf02-fdfe06658192)

joshjavier committed:
Fix download link not working in Edge and Chrome

Previously, the `src` attribute of the image previews are using object
URLs created from the selected image, which are immediately revoked when
the images are loaded on the DOM. This breaks the download link because
the blob that is being referenced by its `href` attribute has already
been cleared from memory.

This commit doesn't revoke the object URL on image load, but only when a
new image is being compressed.

The `loadImage()` function is also refactored so it only runs
`canvas.toBlob()` when loading the compressed version of an image. When
loading the original version of an image, it skips drawing onto the
canvas and simply sets the `src` attribute of the `image-1` slot using
the selected file's data URL.
